### DISCUSSION TITLE
Linux Headless Support

### DISCUSSION THREAD

**Discussion by simeononsecurity** [rustdesk/rustdesk#1326] [Feature Request]
One of the biggest things that TeamViewer has that no other alternative has is that TeamViewer can be installed on headless linux systems. Accessing the command line of a linux system remotely and securely is a every day task for system administrators. Please integrate this as a feature.

**Comment by rustdesk** [comment]
How does it work? can you elaborate more?

**Reply by tlqtangok** [reply]
I also have the same issue.  I can use pure command line version of rustdesk, run on ubuntu-server or some OS that not support GUI.

**Comment by simeononsecurity** [comment]
https://community.teamviewer.com/English/kb/articles/4352-install-teamviewer-on-linux-without-graphical-user-interface
https://community.teamviewer.com/English/kb/articles/6318-install-teamviewer-on-linux
https://community.teamviewer.com/English/kb/articles/4256-use-teamviewer-on-headless-systems

_Framebuffer console without Xorg
TeamViewer can be used to access machines without a graphical desktop environment (Xorg, Wayland). TeamViewer lets you connect to the foreground console (as indicated by fgconsole). In other words, the VT (virtual terminal) which is currently visible on the screen – or would be visible if a screen was attached._

**Comment by mbennette68** [comment]
Yea, I noticed that the other day when I tried to remote into a headless machine. Error generated was no display found or something to that effect. 
I normally use [dwservice.net](https://www.dwservice.net/en/) which does seem to support headless remoting but sometimes audio playback does not work with older Linux distros.
I would say, adding Remote capabilities in to headless machines is a must.

**Comment by jficz** [comment]
This is not "headless" imho. Though it might be useful, it is much more secure and well supported to use `ssh` to connect to a "console" if console access is all one needs.

By "headless" access I imagine headless graphical session which I connect to, i.e. no physical display attached to the server but an X/Wayland session still running.

**Reply by simeononsecurity** [reply]
Yes however, switching to multiple systems using different software is cumbersome.
Additionally, the need for port forwarding. Additionally, port forwarding isn't always advisable. 
You can say the same things about RDP as you can SSH for the purposes of remote management. This tool fulfills a different need. Though a more niche and specific one.

**Reply by jficz** [reply]
I understand the use case and have nothing against it, except it being called "headless" ;)

**Comment by tcoombs** [comment]
I have a couple of Raspberry Pi's that are running Rustdesk which do not have a monitor attached. The get around the problem of "no display found" I have attached a dummy HDMI adapter to each machine. My Raspberry Pi's now think they have a monitor attached and Rustdesk now works fine. The dummy HDMI adapters only cost a few $ from Amazon, etc.

**Comment by deajan** [comment]
I tend to use meshcentral for headless linux administration. Indeed, having rustdesk doing the same would be great.

**Comment by davidshq** [comment]
This is now supported in the latest version of RustDesk, is it not? I see the option to check to allow access to headless linux systems.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/wiki/Headless-Linux-Support

> `sudo rustdesk --option allow-linux-headless Y`

**Reply by shoeper** [reply]
This is nice and I appreciate the effort.

So far it does not work very well though. And having to enable it via UI isn't nice as well.

Chrome Remote Desktop "just works". Go to https://remotedesktop.google.com/u/1/headless , copy command, paste via SSH and login via remotedesktop.google.com. It would be really nice if it would be possible to install RustDesk via SSH, get the ID/or IP and password and just login without the need of a desktop access, beforehand. Chrome Remote Desktop then asks which Desktop to use in case there are multiple and "just works".

**Reply by L-U-C-K-Y** [reply]
> This is nice and I appreciate the effort.
> 
> So far it does not work very well though. And having to enable it via UI isn't nice as well.
> 
> Chrome Remote Desktop "just works". Go to https://remotedesktop.google.com/u/1/headless , copy command, paste via SSH and login via remotedesktop.google.com. It would be really nice if it would be possible to install RustDesk via SSH, get the ID/or IP and password and just login without the need of a desktop access, beforehand. Chrome Remote Desktop then asks which Desktop to use in case there are multiple and "just works".

I agree, is this possible? This would help our workflow a lot

**Reply by L-U-C-K-Y** [reply]
@rustdesk friendly bump: we are very happy with RustDesk, this is the only feature we would lack. 

> It would be really nice if it would be possible to install RustDesk via SSH, get the ID/or IP and password and just login without the need of a desktop access, beforehand.

**Reply by L-U-C-K-Y** [reply]
I'm willing to fund this issue and created a separate discussion (wanted to create an issue for polar.sh) here: https://github.com/rustdesk/rustdesk/discussions/8418

We can close that and continue with an issue, however suits you best

**Reply by rustdesk** [reply]
This is not a small fund (>10k$), because it is actually a big project. I can be 99.9% sure you can not find a dev to do it. :(
